### PR TITLE
add cities tables and seeds

### PR DESCRIPTION
**Pull request by khaledkzy**

### CODE REVIEW

**Comment by islam95** [comment]
Is this still in progress or I can merge it?

File: server/dbClients/citiesDB.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+function getCities () {
+  return knex.select().from('cities')
+};
+
+function addCity (data) {
+  return knex.table('cities').insert({
+    city_id: data.city_id,
+    city_name: data.city_name
+  })
+};
```
Comment (DwinaTech) [inline server/dbClients/citiesDB.js]: try to make data coming from client or postman camelcase like: data.cityId, data.cityName

File: server/dbClients/citiesDB.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+function editCity (id, data) {
+  return knex.table('cities')
+    .where('city_id', '=', id)
+    .update({
+      city_id: data.city_id,
+      city_name: data.long_name
+    })
+};
```
Comment (DwinaTech) [inline server/dbClients/citiesDB.js]: ditto

File: server/dbClients/citiesDB.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+function deleteCity (cityId) {
+  return knex.table('cities')
+    .where('city_id', '=', cityId)
```
Comment (DwinaTech) [inline server/dbClients/citiesDB.js]: Good

File: server/dbClients/citiesDB.js [hunk 4]
```diff
@@ -19,11 +19,11 @@ function getCity (id) {
 }
 
 function editCity (id, data) {
-  return knex.table('cities')
-    .where('city_id', '=', id)
+  return knex.table('cities').then(console.log)
```
Comment (DwinaTech) [inline server/dbClients/citiesDB.js]: remove the console

File: server/routes/api/cities/index.js [hunk 5]
```diff
@@ -23,12 +23,15 @@ router.post('/', async (req, res) => {
 })
 
 router.put('/', async (req, res) => {
+
   const data = req.body
   try {
-    const response = await db.editCity(data)
-    res.status(200).json(response)
+    await db.editCity(data)
+    res.status(200).json({ success: true })
```
Comment (DwinaTech) [inline server/routes/api/cities/index.js]: return the response

File: server/routes/api/cities/index.js [hunk 6]
```diff
@@ -23,12 +23,15 @@ router.post('/', async (req, res) => {
...
   } catch (error) {
-    res.status(502).json(error)
+    console.log(error)
```
Comment (DwinaTech) [inline server/routes/api/cities/index.js]: remove the console

File: server/dbClients/citiesDB.js [hunk 7]
```diff
@@ -18,8 +18,8 @@ function getCity (id) {
     .where('city_id', '=', id)
 }
 
-function editCity (id, data) {
-  return knex.table('cities').then(console.log)
+function editCity (data) {
+  return knex.table('cities')
     .where('city_id', '=', data.city_id)
```
Comment (DwinaTech) [inline server/dbClients/citiesDB.js]: use camelcase

File: server/dbClients/citiesDB.js [hunk 8]
```diff
@@ -18,8 +18,8 @@ function getCity (id) {
...
     .update({
       city_id: data.city_id,
```
Comment (DwinaTech) [inline server/dbClients/citiesDB.js]: ditto

File: server/routes/api/cities/index.js [hunk 9]
```diff
@@ -23,22 +23,19 @@ router.post('/', async (req, res) => {
 })
 
 router.put('/', async (req, res) => {
-
   const data = req.body
   try {
-    await db.editCity(data)
-    res.status(200).json({ success: true })
+    const response = await db.editCity(data)
+    res.status(200).json(response)
```
Comment (DwinaTech) [inline server/routes/api/cities/index.js]: good

File: server/routes/api/cities/index.js [hunk 10]
```diff
@@ -23,22 +23,19 @@ router.post('/', async (req, res) => {
...
+    res.status(502).json(error)
   }
 })
 
 router.delete('/', async (req, res) => {
-  const { cityId } = req.query
+  const data = req.body
   try {
-    const response = await db.deleteCity(cityId)
+    const response = await db.deleteCity(data.city_id)
```
Comment (DwinaTech) [inline server/routes/api/cities/index.js]: use camelcase

**Pull request merged**